### PR TITLE
feat: add jest preset

### PR DESCRIPTION
**Pull request by jquense**
built on #27

### CODE REVIEW

File: packages/jest-preset/transforms/babel.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+const babel = require('@babel/core');
```
Comment (jquense) [inline packages/jest-preset/transforms/babel.js]: this is an alternative to babel-jest that is:

- simpler,
- doesn't also try and support babel 6
- handles typescript out of the box if needed

File: packages/jest-preset/transforms/babel.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    return crypto
+      .createHash('md4')
+      .update(
+        JSON.stringify({
+          source,
+          options,
+          version: pkg.version,
+          jestConfig: configString,
+          '@babel/core': babel.version,
+          env: process.env.NODE_ENV || process.env.BABEL_ENV || '',
```
Comment (jquense) [inline packages/jest-preset/transforms/babel.js]: this is mostly the same as how babel-loader does it

File: packages/jest-preset/transforms/babel.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      );
+    }
+    return babel.transform(src, options) || src;
+  },
+
+  getCacheKey(source, filename, configString) {
+    const { options } = babel.loadPartialConfig(opts) || {};
+
+    return crypto
+      .createHash('md4')
```
Comment (taion) [inline packages/jest-preset/transforms/babel.js]: what the heck is md4?

**Comment by taion** [comment]
ok i didn't review super closely

File: packages/jest-preset/transforms/styles.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+  process() {
+    return 'module.exports = {};';
+  },
+  getCacheKey() {
+    return 'style-transition';
```
Comment (taion) [inline packages/jest-preset/transforms/styles.js]: "transition"?

Comment (jquense) [reply packages/jest-preset/transforms/styles.js]: Yeah that's weird. I copied it from CRA. What does that even mean, I'll fix

**Pull request merged**